xknx/devices/switch.py

```diff
@@ -42,6 +42,11 @@ def _iter_remote_values(self):
         """Iterate the devices RemoteValue classes."""
         yield self.switch
 
+    @property
+    def unique_id(self):
+        """Return unique id for this device."""
+        return f"switch_{hash(self.switch.group_address)}"
```

Comment:
Is there I understood or good reason not to use state here? I mean all the other use state.

Reply:
Not all switches have a state_address. But all switches have a switch_address. (Check the schema of HA - the switch GA is the only required parameter here)

Reply:
Right. I just saw comment on discord.

I wonder if we should prepare for a installation prefix to the unique I'd:s. Might not be worth it.

Reply:
Well right now you cannot have more than one gateway (and therefore installation) in HA if I recall correctly.

Reply:
Right, but we could solve that quite nicely with config entries for the gateway configuration. Obviously need some more work, but it would move us closer to best practice..

Ps.. it would also allow reload of the integration without restart when you add/remove entities in yaml.

Reply:
https://github.com/XKNX/xknx/issues/238

Reply:
PRs are always welcome :dancers:

Reply:
Will see if I get the itch. I tend to spread my time too thin on the integrations I'm involved in anyway, so why not a other :)

Reply:
Haha :-D. I saw you have been dealing with config flows in the past, I might come back to you if I need some help on that - or just ask on HA discord. :-D

Reply:
Absolutely, it's usually fun untill you get an angry hoard of users complaining about having to adjust to a breaking change. :)

Reply:
Haha yeah. I got the hang of that already xD.